aws/session/session.go

```diff
@@ -81,10 +82,16 @@ func New(cfgs ...*aws.Config) *Session {
 				r.Error = err
 			})
 		}
+
 		return s
 	}
 
-	return deprecatedNewSession(cfgs...)
+	s := deprecatedNewSession(cfgs...)
+	if len(envCfg.CSMEnabled) != 0 {
```

Comment:
Does this need to be a string? A boolean would be much more natural. like `EnableSharedConfig`.